query/src/main/java/io/keen/client/java/KeenQueryClient.java

```diff
@@ -0,0 +1,704 @@
+                .withPercentile(percentile)
+                .build();
+        QueryResult result = execute(queryParams, timeframe);
+        return queryResultToDouble(result);
+    }
+
+    /**
+     * Sum Resource query with only the required arguments.
+     * Query API info here: https://keen.io/docs/api/reference/#sum-resource
+     *
+     * @param eventCollection     The name of the event collection you are analyzing.
+     * @param targetProperty     The name of the property you are analyzing.
+     * @param timeframe     The {@link RelativeTimeframe} or {@link AbsoluteTimeframe}.
+     * @return The sum resource response.
+     * @throws IOException If there was an error communicating with the server or
+     * an error message received from the server.
+     */
+    public Double sum(String eventCollection, String targetProperty, Timeframe timeframe) throws IOException  {
+        Query queryParams = new Query.QueryBuilder(QueryType.SUM_RESOURCE)
+                .withEventCollection(eventCollection)
+                .withTargetProperty(targetProperty)
+                .build();
+        QueryResult result =  execute(queryParams, timeframe);
+        return queryResultToDouble(result);
+    }
+
+    /**
+     * Select Unique Resource query with only the required arguments.
+     * Query API info here: https://keen.io/docs/api/reference/#select-unique-resource
+     *
+     * @param eventCollection     The name of the event collection you are analyzing.
+     * @param targetProperty     The name of the property you are analyzing.
+     * @param timeframe     The {@link RelativeTimeframe} or {@link AbsoluteTimeframe}.
+     * @return The select unique query response.
+     * @throws IOException If there was an error communicating with the server or
+     * an error message received from the server.
+     */
+    public QueryResult selectUnique(String eventCollection, String targetProperty, Timeframe timeframe) throws IOException  {
+        Query queryParams = new Query.QueryBuilder(QueryType.SELECT_UNIQUE_RESOURCE)
+                .withEventCollection(eventCollection)
+                .withTargetProperty(targetProperty)
+                .build();
+        QueryResult result = execute(queryParams, timeframe);
+        return result;
+    }
+
+    /**
+     * This is the most flexible way to run a query. Use {@link io.keen.client.java.KeenQueryClient.QueryBuilder} to
+     * build all the query arguments to run the query.
+     *
+     * @param params     The {@link Query} information, including {@link QueryType}, required args, and any optional args.
+     * @param timeframe     The {@link RelativeTimeframe} or {@link AbsoluteTimeframe}.
+     * @return The {@link QueryResult} result.
+     * @throws IOException If there was an error communicating with the server or
+     * an error message received from the server.
+     */
+    public QueryResult execute(Query params, Timeframe timeframe) throws IOException {
```

Comment:
Concern: It feels a little bit weird to me that the timeframe is handled separately from the rest of the query parameters. I know that we want to emphasize its use, but I'm worried that we're sacrificing too much in terms of elegance of the design.

Maybe a compromise is to move timeframe into the `Query` object, but (a) keep it in all of the shortcut methods, (b) keep it in all of the examples in the docs, and (c) log a warning message if `Query.constructQueryArgs` is called and no timeframe is present. What do you think?